ODIN_II/SRC/partial_map.cpp

```diff
@@ -903,128 +903,180 @@ void instantiate_GE(nnode_t* node, operation_list type, short mark, netlist_t* n
  *	Creates the hardware for a shift left or right operation by a constant size.
  *-------------------------------------------------------------------------------------------*/
 void instantiate_shift_left_or_right(nnode_t* node, operation_list type, short mark, netlist_t* netlist) {
-    /* these variables are used in an attempt so that I don't need if cases.  Probably a bad idea, but fun */
-    int width;
-    int i;
-    int shift_size;
-    nnode_t* buf_node;
+    
+    if ( node->related_ast_node->children[1]->type != NUMBERS ) {
```

Comment:
it should never be a number at this point since you should blow constant shifts away (they don't create node). dont do this check

Reply:
So, we need to check it here again since even with the constant shift it goes through the partial map and calls the instance_shift function again.

Reply:
make sure it doesnt, do no create the node

Reply:
change this to an oassert actually. this should crash if it is a number